### PR TITLE
Using chart.js instead of d3

### PR DESCRIPTION
**Pull request by Angamanga**
This pull request makes the following changes:
- Using chart.js instead of  d3 in activity-view
- Updating configs and data formatting to fit chart.js

Testing checklist:
This pr does not change the behaviour of the activity-view, just the library used to display the graphs (and a little bit of the appearance, for example, the bars are now vertical instead of horizontal, but they still display the same thing). The easiest way of testing is to look at the activity-view before merging this pr, take note of the behaviour for different filters, merge the pr and compare the filters. They should reflect the same thing. Or merge this pr and compare the behaviour with another deployment. 

Note: There are known issues with the activity-view (see comment below) and they are not fixed by this PR, they will be handled separately.

- [x] I certify that I ran my checklist

Fixes ushahidi/platform#4258 and ushahidi/platform#429  .

Ping @ushahidi/platform

### CODE REVIEW

**Comment by Angamanga** [comment]
@tuxpiper @webong This PR contains the changes needed to switch from d3 to chart.js. When working on this I found some odd things in the responses from the API. When calling the Post-stats-endpoint with some of the filters used, I get results I cannot figure out. Could we try to clarify or identify if there is a bug?

List of the weirdnesses:
1. When calling the `/api/v3/posts/stats?group_by=&timeline=1&timeline_attribute=created` it looks like I could expect post-count/day a post has been submitted during the time the deployment has existed. For annatest1000.ushahidi.io I get back 2 posts only which is a bit odd since there are 22620 posts in my deployment. 
2. When calling `/api/v3/posts/stats?group_by=tags&timeline=1&timeline_attribute=created` I always get an empty array back. It looks like what the client expects is the number of posts added/day grouped by categories.
3. When calling `/api/v3/posts/stats?group_by=form&timeline=1&timeline_attribute=created` I get the survey names and a total of 1 for each survey, which I suspect is the number of surveys created for each survey but not sure since not all my surveys are included in the response 🤔 . The value I would like is the number of posts/day grouped by survey.
4. When calling `/api/v3/posts/stats?group_by=tags` I also always get an empty array back, but what I expect is the total number of posts grouped by categories.

Either the API has changed over time and we have forgotten to update the activity view or the API has a bug in some way. Not sure if we should merge this before we have figured it out? The behaviour already exists in production so we would not be introducing a new bug, but still, it's annoying to push something half-broken.

**Comment by Angamanga** [comment]
@tuxpiper @webong Ready for review 🎉

**Comment by Ifycode** [comment]
Yay!

File: app/main/activity/bar-chart.directive.js
Comment (tuxpiper) [inline app/main/activity/bar-chart.directive.js:56]: I just smiled at the simplicity of this function :)

Something for the future (not this PR), but maybe let's look for a palette of colors to randomly pick from, for better visual results.

**Review by tuxpiper — COMMENTED** [review]
Leaving the approval up to @webong  . I just added a little comment of mine :) Excited to see this!

**Comment by Angamanga** [comment]
@webong Could you do a final review? 🙏

**Comment by Angamanga** [comment]
@tuxpiper QA passes, just need an approval for the last small changes (using dayjs instead of moment in activity-view)

**Review by tuxpiper — APPROVED** [review]
Yes, I like it 👍  ( @webong  please take look too 🙏 )

**Review by webong — APPROVED** [review]

**Pull request merged**
